code/game/machinery/doors/firedoor.dm

```diff
@@ -83,7 +78,10 @@
 /obj/machinery/door/firedoor/LateInitialize()
 	. = ..()
 	GetMergeGroup(merger_id, allowed_types = merger_typecache)
-
+	merge_group = GetMergeGroup(merger_id, merger_typecache)
+	refresh_shared_turfs(merge_group)
+	check_adjacent_turfs()
+	RegisterSignal(merge_group, COMSIG_MERGER_REFRESH_COMPLETE, .proc/refresh_shared_turfs)
```

Comment:
what you want to do is something like this
https://github.com/tgstation/tgstation/blob/1ce2768199dcaf24e74f2d075c848f9b52d02ede/code/modules/atmospherics/machinery/components/tank.dm#L84-L85
and then register refresh_shared_turfs inside the proc those link to
https://github.com/tgstation/tgstation/blob/1ce2768199dcaf24e74f2d075c848f9b52d02ede/code/modules/atmospherics/machinery/components/tank.dm#L202-L212

This way we actually support joining new groups, and as an upside you don't need to call refresh_shared_turfs manually anymore, if you do all this before the get merge group.

Actually the registration should be done in initialize for reasons, but etc etc etc

Reply:
alright then